### PR TITLE
initial pass at license bot

### PR DESCRIPTION
**Pull request by CJ-Wright**
This would update licenses where possible

### CODE REVIEW

File: conda_forge_tick/migrators.xsh [hunk 1]
```diff
@@ -72,6 +78,47 @@ class PipMigrator(MiniMigrator):
...
+
+    def migrate(self, recipe_dir, attrs, **kwargs):
+        # Use conda build to do all the downloading/extracting bits
+        md = MetaData(recipe_dir)
+        # go into source dir
+        cb_work_dir = provide(md)
+        with indir(cb_work_dir):
+            # look for a license file
+            license_file = next((s for s in os.listdir('.') if
+                                 any([getattr('license', k)() in s for k in ['lower', 'upper', 'title']])), None)
```
Comment (isuruf) [inline conda_forge_tick/migrators.xsh]: Can you also look for `COPYING*`, `COPYRIGHT`, `LICENSE*` and if there are multiple add all of them ?

File: conda_forge_tick/migrators.xsh [hunk 2]
```diff
@@ -72,6 +78,46 @@ class PipMigrator(MiniMigrator):
...
+        return True
+
+    def migrate(self, recipe_dir, attrs, **kwargs):
+        # Use conda build to do all the downloading/extracting bits
+        md = MetaData(recipe_dir)
+        # go into source dir
+        cb_work_dir = provide(md)
+        with indir(cb_work_dir):
+            # look for a license file
+            license_files = [(s for s in os.listdir('.') if s.lower in ['license', 'copying', 'copyright'])]
```
Comment (isuruf) [inline conda_forge_tick/migrators.xsh]: Can you check for files starting with `license` instead of exact match?

Comment (isuruf) [reply conda_forge_tick/migrators.xsh]:
```suggestion
            license_files = [(s for s in os.listdir('.') if s.lower() in ['license', 'copying', 'copyright'])]
```

**Comment by CJ-Wright** [comment]
Note that this needs a conda-smithy PR merge and release.

**Comment by CJ-Wright** [comment]
I think this is getting ready to go @regro/auto-tick. There are some gaps in this functionality, since this doesn't handle the case where the license is not part of the distribution. I think that needs to be a separate PR which actually issues PRs where possible.

**Comment by CJ-Wright** [comment]
One discussion to have is if we want to put this on all migrators, or just some.

File: tests/test_migrators.py [hunk 3]
```diff
@@ -1613,8 +1613,97 @@
...
+    - numpy
+    - matplotlib
+    - colorspacious
+
+test:
+  imports:
+    - viscm
+
+about:
+  license_file: LICENSE
```
Comment (isuruf) [inline tests/test_migrators.py]: Can this be added after `license` entry?

Comment (CJ-Wright) [reply tests/test_migrators.py]: It's a bit more annoying to code up, but if you can put in the needed regex replacement [here](https://github.com/regro/cf-scripts/pull/637/files#diff-65ad1c9390fe432dca7ca211091e3287R122-R124) that would be spectacular.

File: conda_forge_tick/migrators.xsh [hunk 4]
```diff
@@ -72,6 +83,49 @@ class PipMigrator(MiniMigrator):
...
+                    raw = f.read()
+                lines = raw.splitlines()
+                ptn = re.compile(r'(\s*?)' + 'license:')
+                for i, line in enumerate(lines):
+                   m = ptn.match(line)
+                   if m is not None:
+                       break
+                ws = m.group(1)
+                if len(license_files) == 1:
+                    replace_in_file("about:", "about:\n" + ws + f"license_file: {next(iter(license_files))}", 'meta.yaml')
```
Comment (isuruf) [inline conda_forge_tick/migrators.xsh]: This adds only one of the files, right?

Comment (CJ-Wright) [reply conda_forge_tick/migrators.xsh]: yes, the other clause adds the list of files.

Comment (isuruf) [reply conda_forge_tick/migrators.xsh]: :smile: I read that totally wrong

File: conda_forge_tick/migrators.xsh [hunk 5]
```diff
@@ -119,9 +119,9 @@ class LicenseMigrator(MiniMigrator):
                        break
                 ws = m.group(1)
                 if len(license_files) == 1:
-                    replace_in_file("about:", "about:\n" + ws + f"license_file: {next(iter(license_files))}", 'meta.yaml')
+                    replace_in_file(line, line + "\n" + ws + f"license_file: {next(iter(license_files))}", 'meta.yaml')
                 else:
-                    replace_in_file("about:", "about:\n" + ws + f"license_file: {license_files}", 'meta.yaml')
+                    replace_in_file(line, line + "\n" + ws + f"license_file: {license_files}", 'meta.yaml')
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]: Awesome! Thank you! Can you update the test?

**Pull request merged**